cmd/cli/main.go

```diff
@@ -115,24 +115,26 @@ func main() {
 						}
 					}
 
+					runner.EndRun()
+
 					if proofmode {
-						runner.EndRun()
 						if err := runner.FinalizeSegments(); err != nil {
 							return fmt.Errorf("cannot finalize segments: %w", err)
 						}
-						trace, memory, err := runner.BuildProof()
-						if err != nil {
-							return fmt.Errorf("cannot build proof: %w", err)
-						}
-						if traceLocation != "" {
-							if err := os.WriteFile(traceLocation, trace, 0644); err != nil {
-								return fmt.Errorf("cannot write relocated trace: %w", err)
-							}
+					}
+
+					trace, memory, err := runner.BuildProof()
+					if err != nil {
+						return fmt.Errorf("cannot build proof: %w", err)
+					}
```

Comment:
In line with the previous comments, we shouldn't do this unless specified by the user, so it's good to have a flag here. Either proof mode is on or they asked for a trace.

Reply:
done